examples/classification/plot_classification_probability.py

```diff
@@ -3,93 +3,240 @@
-        C=C, penalty="l2", solver="saga", max_iter=10000
+    "Logistic regression\n(C=0.01)": LogisticRegression(C=0.1),
+    "Logistic regression\n(C=1)": LogisticRegression(C=100),
+    "Gaussian Process": GaussianProcessClassifier(kernel=1.0 * RBF([1.0, 1.0])),
+    "Logistic regression\n(RBF features)": make_pipeline(
+        Nystroem(kernel="rbf", gamma=5e-1, n_components=50, random_state=1),
+        LogisticRegression(C=10),
     ),
-    "L2 logistic (OvR)": OneVsRestClassifier(
-        LogisticRegression(C=C, penalty="l2", solver="saga", max_iter=10000)
+    "Gradient Boosting": HistGradientBoostingClassifier(),
+    "Logistic regression\n(binned features)": make_pipeline(
+        KBinsDiscretizer(n_bins=5),
+        PolynomialFeatures(interaction_only=True),
+        LogisticRegression(C=10),
+    ),
+    "Logistic regression\n(interacting spline features)": make_pipeline(
+        SplineTransformer(n_knots=5),
+        PolynomialFeatures(interaction_only=True),
+        LogisticRegression(C=10),
     ),
-    "Linear SVC": SVC(kernel="linear", C=C, probability=True, random_state=0),
-    "GPC": GaussianProcessClassifier(kernel),
 }
 
+# %%
+# Plotting the decision boundaries
+# --------------------------------
+#
+# For each classifier, we plot the per-class probabilities on the first three
+# columns and the probabilities of the most likely class on the last column.
+
 n_classifiers = len(classifiers)
+scatter_kwargs = {
+    "s": 25,
+    "marker": "o",
+    "linewidths": 0.8,
+    "edgecolor": "k",
+    "alpha": 0.7,
+}
+y_unique = np.unique(y)
 
+# Ensure legend not cut off
+mpl.rcParams["savefig.bbox"] = "tight"
 fig, axes = plt.subplots(
     nrows=n_classifiers,
-    ncols=len(iris.target_names),
-    figsize=(3 * 2, n_classifiers * 2),
+    ncols=len(iris.target_names) + 1,
+    figsize=(4 * 2.2, n_classifiers * 2.2),
 )
+evaluation_results = []
+levels = 100
 for classifier_idx, (name, classifier) in enumerate(classifiers.items()):
-    y_pred = classifier.fit(X, y).predict(X)
-    accuracy = accuracy_score(y, y_pred)
-    print(f"Accuracy (train) for {name}: {accuracy:0.1%}")
-    for label in np.unique(y):
+    y_pred = classifier.fit(X_train, y_train).predict(X_test)
+    y_pred_proba = classifier.predict_proba(X_test)
+    accuracy_test = accuracy_score(y_test, y_pred)
+    roc_auc_test = roc_auc_score(y_test, y_pred_proba, multi_class="ovr")
+    log_loss_test = log_loss(y_test, y_pred_proba)
+    evaluation_results.append(
+        {
+            "name": name.replace("\n", " "),
+            "accuracy": accuracy_test,
+            "roc_auc": roc_auc_test,
+            "log_loss": log_loss_test,
+        }
+    )
+    for label in y_unique:
         # plot the probability estimate provided by the classifier
         disp = DecisionBoundaryDisplay.from_estimator(
             classifier,
-            X,
+            X_train,
             response_method="predict_proba",
```

Comment:
This is a pitfall of this method. By default, it would use `decision_function` which is much harder to interpret in my opinion (especially when comparing different model classes).

I think we should change the "auto" policy to use favor `predict_proba` when available but this should rather be done in a dedicated follow-up PR.

Reply:
BTW, if we use `response_method="predict_proba"`, maybe the `vmin=0` and `vmax=1` parameters could be set automatically to make the use `DecisionBoundaryDisplay` terser.

Reply:
It looks indeed like a good suggestion.